devicetypes/zooz/zooz-zse43-tilt-shock-xs-sensor.src/zooz-zse43-tilt-shock-xs-sensor.groovy

```diff
@@ -0,0 +1,386 @@
+		log.warn "Unable to parse: $description"
+	}
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCmd = cmd.encapsulatedCommand(commandClassVersions)
+	if (encapsulatedCmd) {
+		zwaveEvent(encapsulatedCmd)
+	} else {
+		log.warn "Unable to extract encapsulated cmd from $cmd"
+	}
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpNotification cmd) {
+	logDebug "Device Woke Up..."
+	List<String> cmds = []
+
+	cmds += getRefreshCmds()
+	cmds += getConfigureCmds()
+
+	if (!cmds) {
+		cmds << secureCmd(zwave.batteryV1.batteryGet())
+	}
+
+	cmds << secureCmd(zwave.wakeUpV2.wakeUpNoMoreInformation())
+	sendHubCommand(cmds, 150)
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpIntervalReport cmd) {
+	logDebug "Wake Up Interval = ${cmd.seconds} seconds"
+	state.wakeUpInterval = cmd.seconds
+	refreshSyncStatus()
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	Integer val = (cmd.batteryLevel == 0xFF ? 1 : cmd.batteryLevel)
+	if (val > 100) {
+		val = 100
+	}
+	logDebug "Battery is ${val}%"
+	sendEvent(name:"battery", value:val, unit:"%", isStateChange: true)
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.sensorbinaryv2.SensorBinaryReport cmd) {
+	logDebug "${cmd}"
+	if (cmd.sensorType == sensorTypeContact) {
+		sendContactEvent(cmd.sensorValue ? "open" : "closed")
+	}
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	if (cmd.notificationType == accessControl) {
+		if (cmd.event == accessControlOpen) {
+			sendContactEvent("open")
+		} else if (cmd.event == accessControlClosed) {
+			sendContactEvent("closed")
+		} else {
+			logDebug "${cmd}"
+		}
+	} else if (cmd.notificationType == homeSecurity) {
+		sendAccelerationEvent((cmd.event == homeSecurityVibration) ? "active" : "inactive")
+	} else {
+		logDebug "${cmd}"
+	}
+}
+
+void sendContactEvent(String value) {
+	logDebug "Contact is ${value}"
+	sendEvent(name: "contact", value: value)
+	sendContactVibrationEvent(value, device.currentValue("acceleration"))
+}
+
+void sendAccelerationEvent(String value) {
+	logDebug "Acceleration is ${value}"
+	sendEvent(name: "acceleration", value: value)
+	sendContactVibrationEvent(device.currentValue("contact"), value)
+}
+
+void sendContactVibrationEvent(String contactValue, String vibrationValue) {
+	String value
+	switch (getSettingVal("disableEnableSensors")) {
+		case contactOnly:
+			value = contactValue
+			break
+		case vibrationOnly:
+			value = vibrationValue
+			break
+		default:
+			value = "${contactValue}${vibrationValue.capitalize()}"
+	}
+
+	if (device.currentValue("contactVibration") != value) {
+		sendEvent(name: "contactVibration", value: value, displayed: false)
+	}
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.versionv1.VersionReport cmd) {
+	logDebug "${cmd}"
+	sendEvent(name: "firmwareVersion", value: (cmd.applicationVersion + (cmd.applicationSubVersion / 100)))
```

Comment:
Are you intending for this to be used elsewhere later? Is this just for displaying to the user?